core/workflow-core/src/main/scala/edu/uci/ics/amber/core/storage/result/iceberg/IcebergDocument.scala

```diff
@@ -430,4 +448,56 @@ private[storage] class IcebergDocument[T >: Null <: AnyRef](
     }
     filesSize
   }
+
+  private def getUnderlyingFileData(fileScanTasks: Seq[FileScanTask], index: Int): InputStream = {
+    val fileTask = fileScanTasks(index)
+    val file = fileTask.file()
+    val table = catalog.loadTable(TableIdentifier.of(tableNamespace, tableName))
+    table.io().newInputFile(file.path().toString).newStream()
+  }
+
+  override def asInputStream(): InputStream = {
+    val fileScanTasks: Seq[FileScanTask] = {
+      val table = this.catalog.loadTable(TableIdentifier.of(this.tableNamespace, this.tableName))
+      table.refresh()
+      table.newScan().planFiles().iterator().asScala.toSeq
+    }
+
+    if (fileScanTasks.isEmpty) {
+      return new ByteArrayInputStream(Array.emptyByteArray)
+    }
+
+    val pipeIn = new PipedInputStream(Constants.DEFAULT_BUFFER_SIZE)
+    val pipeOut = new PipedOutputStream(pipeIn)
+
+    // Start processing in a separate thread to avoid blocking
+    val processingThread = new Thread(() => {
+      try {
```

Comment:
Is it better to pass the Seq of FileScanTask as the parameter to the Thread, instead of letting the thread access the outer variable?